### PR TITLE
Use Scala's execution context

### PR DESCRIPTION
**Pull request by jvican**

### CODE REVIEW

**Comment by jvican** [comment]
test performance please

**Comment by bloopoid** [comment]
performance test scheduled: 1 job(s) in queue, 1 running.

File: bin/run-benchmarks.sh [hunk 1]
```diff
@@ -106,7 +106,7 @@ main() {
       "-wi 10 -i 5 -f1 -t1 -p project=coursier -p projectName=coursier-repo-test"
       "-wi 15 -i 10 -f1 -t1 -p project=prisma -p projectName=root-test"
       "-wi 5 -i 3 -f1 -t1 -p project=cats -p projectName=cats-test" # compiles hot in 3 minutes
-      "-wi 7 -i 5 -f1 -t1 -p project=scio -p projectName=scio+test"
+      #"-wi 2 -i 3 -f1 -t1 -p project=scio -p projectName=scio+test"
```
Comment (tgodzik) [inline bin/run-benchmarks.sh]: Is this commented out on purpose?

File: frontend/src/main/scala/bloop/bsp/BspServer.scala [hunk 2]
```diff
@@ -245,9 +246,14 @@ object BspServer {
...
-      val externalClientClassesDir = latestState.client.getUniqueClassesDirFor(project)
-      if (externalClientClassesDir == project.genericClassesDir) Task.now(())
-      else Task.eval(Paths.delete(externalClientClassesDir)).executeWithFork
+      try {
+        val externalClientClassesDir = latestState.client.getUniqueClassesDirFor(project)
+        if (externalClientClassesDir == project.genericClassesDir) Task.now(())
+        else Task.fork(Task.eval(Paths.delete(externalClientClassesDir)))
+      } catch {
+        // Can be thrown from `getUniqueClassesDirFor`
+        case _: NoSuchFileException => Task.now(())
```
Comment (tgodzik) [inline frontend/src/main/scala/bloop/bsp/BspServer.scala]: Can't we check if that file exists instead of the try?

**Comment by bloopoid** [comment]
Performance test finished successfully.

Benchmarks is based on merging with master

**Comment by jvican** [comment]
test performance please

**Comment by bloopoid** [comment]
performance test scheduled: 1 job(s) in queue, 1 running.

**Comment by jvican** [comment]
test performance please

**Comment by bloopoid** [comment]
performance test scheduled: 1 job(s) in queue, 1 running.

**Comment by jvican** [comment]
test performance please

**Comment by bloopoid** [comment]
performance test scheduled: 1 job(s) in queue, 1 running.

**Comment by bloopoid** [comment]
Performance test finished successfully.

Benchmarks is based on merging with master

**Comment by jvican** [comment]
test performance please

**Comment by bloopoid** [comment]
performance test scheduled: 1 job(s) in queue, 1 running.

**Comment by bloopoid** [comment]
Performance test finished successfully.

Benchmarks is based on merging with master

**Comment by jvican** [comment]
test performance please

**Comment by bloopoid** [comment]
performance test scheduled: 1 job(s) in queue, 1 running.

**Comment by bloopoid** [comment]
Performance test finished successfully.

Benchmarks is based on merging with master

**Comment by jvican** [comment]
test performance please

**Comment by bloopoid** [comment]
performance test scheduled: 1 job(s) in queue, 1 running.

**Comment by bloopoid** [comment]
Performance test finished successfully.

Benchmarks is based on merging with master

**Comment by jvican** [comment]
test performance please

**Comment by bloopoid** [comment]
performance test scheduled: 1 job(s) in queue, 1 running.

**Comment by jvican** [comment]
test performance please

**Comment by bloopoid** [comment]
performance test scheduled: 1 job(s) in queue, 1 running.

**Comment by jvican** [comment]
test performance please

**Comment by bloopoid** [comment]
performance test scheduled: 1 job(s) in queue, 1 running.

**Comment by jvican** [comment]
test performance please

**Comment by bloopoid** [comment]
performance test scheduled: 1 job(s) in queue, 2 running.

**Comment by jvican** [comment]
test performance please

**Comment by bloopoid** [comment]
performance test scheduled: 1 job(s) in queue, 0 running.

**Comment by jvican** [comment]
test performance please

**Comment by bloopoid** [comment]
performance test scheduled: 1 job(s) in queue, 0 running.

**Comment by bloopoid** [comment]
Performance test finished successfully.

Benchmarks is based on merging with master

File: frontend/src/main/scala/bloop/bsp/BspServer.scala (hunk 2 above)
Comment (jvican) [reply frontend/src/main/scala/bloop/bsp/BspServer.scala]: It's not as trivial as doing a check for existence, that was the first approach I tried and it didn't work, but you're right this falls short. I'll rebase this PR on top of the PR changing lots of things re classes directories and try to come up with a better solution.

**Pull request merged**